h2o-algos/src/main/java/hex/tree/DTree.java

```diff
@@ -709,9 +722,17 @@ public DecidedNode(UndecidedNode n, DHistogram hs[], Constraints cs, GlobalInter
         // which might be because all predictor columns are now constant, or
         // because all responses are now constant.
         _splat = Float.NaN;
-        Arrays.fill(_nids,ScoreBuildHistogram.UNDECIDED_CHILD_NODE_ID);
+        Arrays.fill(_nids, ScoreBuildHistogram.UNDECIDED_CHILD_NODE_ID);
         return;
       }
+      if(cs != null) {
+        int constr = cs.getColumnConstraint(_split._col);
+        if (_tree._checkConstraintConsistency && !cs._dist._family.equals(DistributionFamily.quantile) && nid() != 0 && constr != 0) {
```

Comment:
Can we have a unit test for this?

Reply:
E.g: https://github.com/h2oai/h2o-3/blob/master/h2o-extensions/xgboost/src/test/java/hex/tree/xgboost/XGBoostTest.java#L86

in xgboost its done like this-

Reply:
In GBMTest, another parameter has already been used. The constraint check parameter can be used only for models with constraints, so it only makes sense to test it for some specific use cases. Also, it is computationally expensive, so I created only one test case.